spec/model/post_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject { User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }
+
+  it 'title should be present' do
+    post = Post.new(author: subject.id, title: 'Hello', text: 'This is my first post')
+    post.title = nil
+    expect(post).to_not be_valid
+  end
+
+  it 'comments_counter should be an integer greater than or equal to zero' do
+    post = Post.new(author: subject, title: 'Hello', text: 'This is my first post')
+    post.comments_counter = 'invalid'
+    expect(post).to_not be_valid
+
+    post.comments_counter = -1
+    expect(post).to_not be_valid
+
+    post.comments_counter = 0
+    post.likes_counter = 0
+    expect(post).to be_valid
+  end
+
+  it 'title should not be more than 250 characters' do
+    post = Post.new(author: subject.id, title: 'A' * 251, text: 'This is my first post')
+    expect(post).to_not be_valid
+  end
+
+  it 'likes_counter should be an integer greater than or equal to zero' do
+    post = Post.new(author: subject, title: 'Hello', text: 'This is my first post')
+    post.likes_counter = 'invalid'
+    expect(post).to_not be_valid
+
+    post.likes_counter = -1
+    expect(post).to_not be_valid
+
+    post.likes_counter = 0
+    post.comments_counter = 0
+    expect(post).to be_valid
+  end
+end
```

Comment:
### Required
- [ ] It is a requirement that all custom methods in the models should be tested, so please do so as to meet the requirements.
Methods to test from post model are:
- `recent_comments`
- `update_comments_counter`
- `update_likes_counter`
- `increment_user_posts_counter`

## Note This Is Important But (Optional)
- [ ] Your models were not set up correctly. Not to worry just a small minor change and you can get back on track. As it stands all your methods that are supposed to be in the `like` and `comment` models are in the post. So before you write tests for these methods above, I would advise you to move the particular methods to their respective models to avoid complications as you proceed with your app.

Move
- `update_likes_counter` method to the like model and create a `rspec` file to test the method in the `rspec/models` folder
-  `update_comments_counter` method to the comment model and create a `rspec` file to test the method in the `rspec/models` folder

If you do decide to implement these important changes remember to add callbacks for your methods within the models as it will improve your overall application
e.g

```
after_save  `:method_name`